cookbook/case_studies/house_price_prediction/multiregion_house_price_predictor.py

```diff
@@ -0,0 +1,158 @@
+"""
+Predicting House Prices in Multiple Regions Using an XGBoost Model and Flytekit (Python)
+----------------------------------------------------------------------------------------
+
+"""
+
+# %%
+# Step 1: Importing the Libraries
+# ----------------------------------
+import typing
+
+import flytekit
+import pandas as pd
+from flytekit import Resources, dynamic, task, workflow
+from flytekit.types.file import FlyteFile
+
+try:
+    from .house_price_predictor import (
+        generate_and_split_data,
+        fit,
+        predict,
+    )
+except ImportError:
+    from house_price_predictor import (
+        generate_and_split_data,
+        fit,
+        predict,
+    )
+
+# %%
+# Step 2: Initializing the Variables
+# ----------------------------------
+NUM_HOUSES_PER_LOCATION = 1000
+COLUMNS = [
+    "PRICE",
+    "YEAR_BUILT",
+    "SQUARE_FEET",
+    "NUM_BEDROOMS",
+    "NUM_BATHROOMS",
+    "LOT_ACRES",
+    "GARAGE_SPACES",
+]
+LOCATIONS = [
+    "NewYork_NY",
+    "LosAngeles_CA",
+    "Chicago_IL",
+    "Houston_TX",
+    "Dallas_TX",
+    "Phoenix_AZ",
+    "Philadelphia_PA",
+    "SanAntonio_TX",
+    "SanDiego_CA",
+    "SanFrancisco_CA",
+]
+
+# %%
+# Step 3: Task -- Generating & Splitting the Data for Multiple Regions
+# --------------------------------------------------------------------
+#
+# Train, validation, and test datasets are lists of DataFrames.
+@dynamic(cache=True, cache_version="0.1", limits=Resources(mem="600Mi"))
+def generate_and_split_data_multiloc(
+    locations: typing.List[str],
+    number_of_houses_per_location: int,
+    seed: int,
+) -> (typing.List[pd.DataFrame], typing.List[pd.DataFrame], typing.List[pd.DataFrame]):
+    train_sets = []
+    val_sets = []
+    test_sets = []
+    for loc in locations:
+        _train, _val, _test = generate_and_split_data(
+            loc=loc, number_of_houses=number_of_houses_per_location, seed=seed
+        )
+        train_sets.append(
+            _train,
+        )
+        val_sets.append(
+            _val,
+        )
+        test_sets.append(
+            _test,
+        )
+    return train_sets, val_sets, test_sets
+
+
+# %%
+# Step 4: Dynamic Task -- Training the XGBoost Model for Multiple Regions
+# -----------------------------------------------------------------------
+# (A "Dynamic" Task (aka Workflow) spins up internal workflows)
+#
+# Serialize the XGBoost models using joblib and store the models in dat files.
+@dynamic(cache=True, cache_version="0.1", limits=Resources(mem="600Mi"))
```

Comment:
I think we should combine the parallel fit and parallel predict into one @dynamic task like
I think this is more powerful!

```python
@dynamic
def parallel_fit_predict(...)
for ... in .....:
    m = fit(...)
    v = predict(m)
```
@wild-endeavor Wdyt?

Reply:
we can add it in the comments that we can alternatively have 2 parallel methods, but this basically makes it possible to parallelize an entire set of tasks?

Reply:
sure.

Reply:
Done.

Reply:
Done what?

Reply:
I merged the two dynamic tasks into one.